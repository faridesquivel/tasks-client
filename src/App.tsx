import React from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import AddTask from './screens/AddTask';
import ListTasks from './screens/ListTasks';
import TaskDetail from './screens/TaskDetail';
import SignIn from './screens/SignIn';
import SignUp from './screens/SignUp';
import Navbar from './components/Navbar';
import ProtectedRoute from './components/Routes/ProtectedRoute';
import OpenRoute from './components/Routes/OpenRoute';
import './App.css';
import { RootState } from './store';
import { AuthState } from './store/auth/types';

export function App() {
  const dispatch = useDispatch();
  const authState = useSelector<RootState, AuthState>((state) => state.auth);
  
  const signOut = () => {
    dispatch({ type: 'SIGN_OUT' })
  };
  const isLoggedIn = authState.token === null;
  return (
    <div className="App">
      <BrowserRouter>
        <Navbar isAuthenticated={isLoggedIn} signOut={signOut}/>
        <Switch>
          <OpenRoute exact path='/signUp' component={SignUp}/>
          <OpenRoute exact path='/signIn' component={SignIn}/>
          <ProtectedRoute exact path='/tasks' isAuthenticated={isLoggedIn} component={ListTasks}/>
          <ProtectedRoute exact path='/addTask' isAuthenticated={isLoggedIn} component={AddTask}/>
          <ProtectedRoute exact path='/taskDetail/:tid' isAuthenticated={isLoggedIn} component={TaskDetail}/>
          <Route exact path="/">
            <Redirect to="/tasks" />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
