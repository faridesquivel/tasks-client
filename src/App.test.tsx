import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
 
import App from './App';
 
const mockStore = configureStore([]);
 
describe('App renders connected as React-Redux Component', () => {
  let store;
  let component: any;
 
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
 
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});