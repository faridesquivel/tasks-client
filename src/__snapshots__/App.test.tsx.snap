// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders connected as React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="App"
>
  <div
    className="NavbarContainer"
  >
    <div
      className="Title"
    >
      Tasks App
    </div>
    <div
      className="NavbarItemsContainer"
    >
      <div
        className="NavbarItem"
      >
        <a
          className="Link"
          href="/tasks"
          id="list"
          onClick={[Function]}
        >
          Tasks List
        </a>
      </div>
      <div
        className="NavbarItem"
      >
        <a
          className="Link"
          href="/addTask"
          onClick={[Function]}
        >
          Add New Task
        </a>
      </div>
      <div
        className="NavbarItem"
      >
        <button
          className="Link"
          onClick={[Function]}
        >
          Sign Out
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root makeStyles-topGrid-1 MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          xs=12
        </div>
      </div>
    </div>
  </div>
</div>
`;
